File: Pard.Domain/Entities/Locations/Location.cs
using System;
using Pard.Domain.Entities.Records;

namespace Pard.Domain.Entities.Locations
{
    public class Location
    {
        public Guid Id { get; set; }
        public double Longitude { get; set; }
        public double Latitude { get; set; }
        public string AddressStreet { get; set; }
        public string AddressCity { get; set; }
        public string AddressState { get; set; }
        public string AddressCountry { get; set; }

        public Guid RecordId { get; set; }
        public Record Record { get; set; }
    }
}

File: Pard.Angular/Pard-Application/src/app/records/models/location.ts
import { Marker } from './marker';
export interface Location {
    id?: string;
    latitude: number;
    longitude: number;
    // tslint:disable-next-line:ban-types
    viewport?: Object;
    zoom: number;
    addressCity?: string;
    addressStreet?: string;
    addressState?: string;
    addressCountry?: string;
    marker?: Marker;
    recordId?: string;
}
